src/librustc_resolve/lib.rs

```diff
@@ -2121,7 +2123,13 @@ impl<'a> Resolver<'a> {
                                                             impl_item.span,
                                             |n, s| ResolutionError::TypeNotMemberOfTrait(n, s));
 
-                                        this.visit_ty(ty);
+                                        // We also need a new scope for the associated type
+                                        // specific type parameters.
+                                        let type_parameters =
+                                            HasTypeParameters(&impl_item.generics, ItemRibKind);
```

Comment:
Is `ItemRibKind` the right `RibKind`? Is this the right way to visit these types? I copied what was done for methods a few lines above this.